src/runtime/app.ts

```diff
@@ -40,7 +40,9 @@ function createNitroApp(): NitroApp {
     onError: errorHandler,
   });
 
-  h3App.use(config.app.baseURL, timingMiddleware);
+  if (config.timing) {
```

Comment:
config in this context refers to `runtimeConfig` and not the nitro config.

I agree it would be good to remove the header, but ideally as part of the build and not at runtime (wdyt @pi0?). At the moment the only if/then that happens is this:

https://github.com/unjs/nitro/blob/5831707ca653a1383fe52a0212ec9247424e8601/src/rollup/config.ts#L137-L139

Reply:
it feels like it would be best to implement timing as a nitro plugin so it doesn't need to be present in the `app` file at all.

Reply:
@danielroe Thanks for the hint, I've moved it to plugin, is this what you suggested ?

Reply:
exactly! ❤️